### PR TITLE
Changing to pointer interface to store uniform or nonuniform grid.

### PR DESCRIPTION
**Pull request by MFraters**
I only tested if it compiled, I didn't check yet if if the uniform grid give a sensible output or optimized the code.

### CODE REVIEW

File: source/geometry_model/ellipsoidal_chunk.cc [hunk 1]
```diff
@@ -856,11 +923,13 @@ namespace aspect
                       point_lists[i_topo].resize(temp_coordinate_size,std::vector<double>(2,0));
                       
                       for(unsigned int i_coord = 0; i_coord < temp_coordinate_size; i_coord++)
+                      {
                         point_lists[i_topo][i_coord] = Utilities::string_to_double(Utilities::split_string_list(temp_cooridnates[i_coord],':'));
-                      Assert(temp_topography.size() == 2,ExcMessage ("The given coordinate '" + temp_cooridnates[i_coord] + "' is not correct. "
+                        Assert(temp_topography.size() == 2,ExcMessage ("The given coordinate '" + temp_cooridnates[i_coord] + "' is not correct. "
```
Comment (anne-glerum) [inline source/geometry_model/ellipsoidal_chunk.cc]: change `temp_cooridnates` to `temp_coordinates`

File: source/geometry_model/ellipsoidal_chunk.cc [hunk 2]
```diff
@@ -219,13 +245,13 @@ namespace aspect
 
         				if(In2dPolygon(p1, point_lists[i]))
         				{
-        					data[i_long][i_lat] = topography_values[i];
+        					data[i_long * uniform_grid_number_data_points[1] + i_lat] = topography_values[i];
```
Comment (anne-glerum) [inline source/geometry_model/ellipsoidal_chunk.cc]: if `i_long` has reached `uniform_grid_number_data_points[0]` and `i_lat` has reached `uniform_grid_number_data_points[1]` then it seems to me you're writing to `data` outside of the range `uniform_grid_number_data_points[0] * uniform_grid_number_data_points[1]`

File: include/aspect/geometry_model/ellipsoidal_chunk.h [hunk 3]
```diff
@@ -79,9 +80,11 @@ namespace aspect
         
         void set_uniform_grid_number_data_points(std::vector<double>& uniform_grid_number_data_points);
```
Comment (anne-glerum) [inline include/aspect/geometry_model/ellipsoidal_chunk.h]: move up to other set functions

**Comment by anne-glerum** [comment]
Nice. Maybe astyle?

File: source/geometry_model/ellipsoidal_chunk.cc (hunk 2 above)
Comment (MFraters) [inline source/geometry_model/ellipsoidal_chunk.cc]: if i_long is 3 and i_lat is 5, then there are a total of 3*5 = 15 points. The loop of i_lat goes from 0 to 4 and does that 3 times which is again 15 points. What am I missing?

**Comment by MFraters** [comment]
Should I also reorder the functions, of keep that for later?

File: source/geometry_model/ellipsoidal_chunk.cc (hunk 2 above)
Comment (anne-glerum) [inline source/geometry_model/ellipsoidal_chunk.cc]: No you're right, sorry

**Comment by MFraters** [comment]
hmm, this works quite well 👍

**Comment by anne-glerum** [comment]
yes! Merging your stuff with my read file stuff now

**Pull request merged**